bin/runtime/src/lib.rs

```diff
@@ -383,12 +391,82 @@ impl<I: From<u32>> ::frame_support::traits::Get<I> for SessionPeriod {
 impl pallet_session::Config for Runtime {
     type Event = Event;
     type ValidatorId = <Self as frame_system::Config>::AccountId;
-    type ValidatorIdOf = ConvertInto;
+    type ValidatorIdOf = pallet_staking::StashOf<Self>;
     type ShouldEndSession = pallet_session::PeriodicSessions<SessionPeriod, Offset>;
     type NextSessionRotation = pallet_session::PeriodicSessions<SessionPeriod, Offset>;
-    type SessionManager = pallet_aleph::AlephSessionManager<Self>;
+    type SessionManager = pallet_session::historical::NoteHistoricalRoot<Self, Staking>;
     type SessionHandler = <SessionKeys as OpaqueKeys>::KeyTypeIdProviders;
     type Keys = SessionKeys;
+    type WeightInfo = pallet_session::weights::SubstrateWeight<Runtime>;
+}
+
+impl pallet_session::historical::Config for Runtime {
+    type FullIdentification = pallet_staking::Exposure<AccountId, Balance>;
+    type FullIdentificationOf = pallet_staking::ExposureOf<Runtime>;
+}
+
+parameter_types! {
+    pub const BondingDuration: EraIndex = 14;
+    pub const SlashDeferDuration: EraIndex = 13;
+    pub const MaxNominatorRewardedPerValidator: u32 = 512;
+    pub const OffendingValidatorsThreshold: Perbill = Perbill::from_percent(33);
```

Comment:
What is the consequence of setting this to 33? What will happen if this threshold is reached? (And what really counts towards this threshold?)

Reply:
A new era will be forced after the session in which the threshold has been reached. (Slaching)

Reply:
Yes, what does it mean that a new era will be forced?

Reply:
That a new era with new validators that were elected for it will start in the next session and not in possible further session as planned before.

Reply:
Ok, why would we want that? I mean, it's not like forcing a new era will help with anything if finality is stalled. And we don't want the eras to randomly change because something unexpected happened, or at least not till we control the validators.

One general question is: is modifying these constants possible in an update after we launch the pallet?

Reply:
It's possible to modify in an update.

Eras won't randomly change since for a validator to get kicked out of the validators list a sudo call is needed.

Reply:
How do we modify these constants? Some kind of custom storage migration?

Reply:
Oh, or are these contants only in the runtime and not in the storage?